runtime/drivers/notifier.go

```diff
@@ -0,0 +1,49 @@
+package drivers
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+)
+
+// Notifier sends notifications.
+type Notifier interface {
+	SendAlertStatus(s *AlertStatus, spec NotifierSpec) error
+	SendScheduledReport(s *ScheduledReport, spec NotifierSpec) error
+}
```

Comment:
Considering the generic suggestion above, the usage of this could be refactored to:
```go
# pseudo-code
for _, notifier := range alert.Spec.Notifiers { // Could be done in parallel
  connector := runtime.AcquireHandle(notifier.Connector)
  notifier := connector.AsNotifier(notifier.Properties) // Notice the recipient properties are passed here
  notifier.SendAlertStatus(alertStatus)
}
```